templates/server-statefulset.yaml

```diff
@@ -75,6 +87,11 @@ spec:
             {{- include "vault.extraEnvironmentVars" .Values.server | nindent 12 }}
             {{- include "vault.extraSecretEnvironmentVars" .Values.server | nindent 12 }}
           volumeMounts:
```

Comment:
Vault's chart already provides a mechanism for adding custom volumes and mounts via values. Seems like we should be using that and creating the secret separately. 

https://github.com/hashicorp/vault-helm/blob/master/templates/_helpers.tpl#L94-L109
https://github.com/hashicorp/vault-helm/blob/master/templates/_helpers.tpl#L152-L172

What you have here is perfectly reasonable but I'd rather keep this fork as tight with upstream as possible in order to eventually get 100% in sync and drop the fork.

Reply:
yeah i agree with you about keeping it in sync. Originally I had the secrets/cm defined alone, no added mounts or volumes, are we talking about going back to that approach or is this something slightly different? 

Also, if we are using a "per installation" name for the secrets/cm, the helm user will have a tougher time identifying the correct value to put for their volume source 
https://github.com/TakeScoop/vault-helm/pull/6/files#diff-60ca1594dfcfe4f0d22db67a9583d9feR46

Reply:
> Originally I had the secrets/cm defined alone, no added mounts or volumes, are we talking about going back to that approach or is this something slightly different?

Not clear what that means. 

I'm talking about creating a secret outside of the chart and then passing in values so that it gets mounted to the container. There is no need to modify the chart to support TLS, it fully supports mounting TLS material from a secret as-is.

Reply:
> > Originally I had the secrets/cm defined alone, no added mounts or volumes, are we talking about going back to that approach or is this something slightly different?
> 
> Not clear what that means.

This PR originally was only creating a secret and a configmap, no volumes or mounts. You're saying don't add anything to vault-helm? Happy to do this, but I had thought we wanted helm to control all of the resources for the vault installation.

Reply:
> I had thought we wanted helm to control all of the resources for the vault installation.

Not really. If it's convenient, it's fine to use Helm, but there's no particular reason Helm _must_ be responsible for a Vault secret.

> This PR originally was only creating a secret and a configmap, no volumes or mounts. You're saying don't add anything to vault-helm?

I guess your sticking point was that the volume/mount values accept a static secret name? That's why I'm suggesting creating the secret/CM externally, it results in the least amount of configuration changes.

Reply:
cool i'm happy with creating the resources in base installer. I'll close this pr. thanks!

Reply:
I would think in terms of a world where the base installer is just a composition of "add on" Terraform modules. It would be no problem to combine `helm_chart` resources with `kubernetes_secret` resources. It so happens that our current install strategy makes that distinction somewhat more apparent. What's ultimately important is that configurations be automated, which in practice means they should be managed with Terraform, and further implementation details are irrelevant.